cms/bundles/forms.py

```diff
@@ -76,6 +87,69 @@ def _has_datasets(self) -> bool:
 
         return has_datasets
 
+    def _get_dataset_title_from_form_data(self, dataset_id: str) -> str:
+        """Get dataset title from form data based on dataset ID."""
+        for form in self.formsets["bundled_datasets"].forms:
+            if (
+                form.is_valid()
+                and not form.cleaned_data.get("DELETE")
+                and (dataset := form.cleaned_data.get("dataset"))
+                and dataset.namespace == dataset_id
+            ):
+                return str(dataset.title)
+        return dataset_id
+
+    def _check_bundle_contents_approval_status(self) -> list[str]:
+        """Check bundle contents and return list of non-approved datasets."""
+        client = BundleAPIClient()
+        datasets_not_approved: list[str] = []
+
+        # Ensure bundle_api_id is not None
+        if not self.instance.bundle_api_id:
+            return datasets_not_approved
+
+        try:
+            response = client.get_bundle_contents(self.instance.bundle_api_id)
+
+            # Check each content item in the bundle
+            for item in response.get("contents", []):
+                item_state = item.get("state", "unknown")
+
+                if item_state != BundleStatus.APPROVED.value:
+                    # Find the corresponding dataset title from the metadata
+                    metadata = item.get("metadata", {})
+                    dataset_id = metadata.get("dataset_id", "unknown")
+                    dataset_title = self._get_dataset_title_from_form_data(dataset_id)
+                    datasets_not_approved.append(f"{dataset_title} (state: {item_state})")
```

Comment:
What if we have 2 datasets with different editions? The titles would be the same. Include the edition as well?

Reply:
4319c34e